src/main/java/org/humancellatlas/ingest/bundle/BundleManifestService.java

```diff
@@ -0,0 +1,115 @@
+
+        Map<String, Set<MetadataDocument>> hits = new HashMap<>();
+
+        long fileStartTime = System.currentTimeMillis();
+        Iterator<BundleManifest> iterator = allManifestsIterator();
+
+        while(iterator.hasNext()) {
+            BundleManifest bundleManifest = iterator.next();
+            documents.forEach(document -> {
+                String documentUuid = document.getUuid().getUuid().toString();
+                String bundleUuid = bundleManifest.getBundleUuid();
+                EntityType documentType = document.getType();
+                if(entityMapFromManifest(documentType, bundleManifest).containsKey(documentUuid)){
+                    if(hits.containsKey(bundleUuid)) {
+                        hits.get(bundleUuid).add(document);
+                    } else {
+                        hits.put(bundleUuid, new HashSet<>(Collections.singletonList(document)));
+                    }
+                }
+            });
+        }
+
+        long fileEndTime = System.currentTimeMillis();
+        float fileQueryTime = ((float)(fileEndTime - fileStartTime)) / 1000;
+        String fileQt = new DecimalFormat("#,###.##").format(fileQueryTime);
+        log.info("Finding bundles to update took {}s", fileQt);
+        log.info("documentsToUpdate: {}, bundlesToUpdate:{}", documents.size(), hits.keySet().size());
+        return hits;
+    }
+
+    private Iterator<BundleManifest> allManifestsIterator() {
+        Iterator<BundleManifest> manifestsIterator = new Iterator<BundleManifest>() {
+            Pageable pageable = new PageRequest(0, 5000);
+            Page<BundleManifest> pagedBundleManifests = null;
+            Queue<BundleManifest> bundleManifests;
+
+            private void fetch(Pageable pageable){
+                pagedBundleManifests = bundleManifestRepository.findAll(pageable);
+                bundleManifests = new LinkedList<>(pagedBundleManifests.getContent());
+            }
+
+            @Override
+            public boolean hasNext() {
+                return (pagedBundleManifests == null || bundleManifests.size() > 0 || pagedBundleManifests.hasNext());
+            }
+
+            @Override
+            public BundleManifest next() {
+                BundleManifest bundleManifest = null;
+
+                if (pagedBundleManifests == null){
+                    fetch(pageable);
+
+                }
+
+                if(bundleManifests.size() == 0 && pagedBundleManifests.hasNext()) {
+                    fetch(pagedBundleManifests.nextPageable());
+                }
+
+                if(bundleManifests.size() > 0) {
+                    bundleManifest = bundleManifests.remove();
+                }
+
+                return bundleManifest;
+            }
+
+        };
+
+        return manifestsIterator;
+    }
+
+
+    private Map<String, Collection<String>> entityMapFromManifest(EntityType entityType, BundleManifest bundleManifest) {
```

Comment:
This is pretty horrific - I'm sure I've seen this exact bit of code elsewhere too. A better solution may be to push this logic into bundleManifest, or if it does exist also elsewhere at least to centralize it in one place. However, this can probably be left to a later time.

Reply:
Hmm this file shouldn't actually be included in this change.This is unused in this hotfix. Let's see how we can improve this in dev.